### PR TITLE
chore: convert json scenarios to toml format

### PR DESCRIPTION
**Pull request by samypr100**
Closes https://github.com/astral-sh/packse/issues/177 , https://github.com/astral-sh/packse/issues/18

* toml files where automatically generated the from the json files using a basic script
  * Used the filename as the folder name and the scenario name as the filename
* Prettier on the generated ones for consistency

### CODE REVIEW

**Comment by samypr100** [comment]
Here's the basic script I used, so we can regen if needed. For prettier I used the [prettier-plugin-toml](https://github.com/un-ts/toml-tools/tree/master/packages/prettier-plugin-toml) as a separate follow up command.

<details><summary>json_scenarios_to_toml.py</summary>
<p>

```python
import json
import re
from pathlib import Path

import toml

scenarios_path = Path(__file__).parent / "scenarios"

for path in scenarios_path.glob("*.json"):

    with path.open("r") as f:
        json_scenarios = json.load(f)

    for json_scenario in json_scenarios:
        # Use the name key as the filename
        file_path = path.parent / re.sub(r'[\s-]+', '_', path.stem.lower()) / f"{json_scenario['name']}.toml"
        file_path.parent.mkdir(parents=True, exist_ok=True)

        toml_data = {
            'name': json_scenario['name'],
            'description': json_scenario['description'],
            'root': json_scenario['root'],
            'packages': json_scenario['packages'],
            'expected': json_scenario['expected']
        }

        with file_path.open('w', encoding="utf-8") as toml_file:
            toml.dump(toml_data, toml_file)
            print(f"Created: {file_path}")
```

</p>
</details>

**Comment by notatallshaw** [comment]
These are so much nicer to read.

File: scenarios/excluded/dependency-excludes-non-contiguous-range-of-compatible-versions.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+description = "There is a non-contiguous range of compatible versions for the requested package `a`, but another dependency `c` excludes the range. This is the same as `dependency-excludes-range-of-compatible-versions` but some of the versions of `a` are incompatible for another reason e.g. dependency on non-existent package `d`."
+
+[root]
+requires = ["a", "b>=2.0.0,<3.0.0", "c"]
+
+[expected]
+satisfiable = false
+explanation = "Only the `2.x` versions of `a` are available since `a==1.0.0` and `a==3.0.0` require incompatible versions of `b`, but all available versions of `c` exclude that range of `a` so resolution fails."
+
+[packages.a.versions."1.0.0"]
```
Comment (zanieb) [inline scenarios/excluded/dependency-excludes-non-contiguous-range-of-compatible-versions.toml:11]: Can we use nesting for package versions? Is that easier to read?

**Comment by zanieb** [comment]
Exciting! Thanks!

File: scenarios/excluded/dependency-excludes-non-contiguous-range-of-compatible-versions.toml (hunk 1 above)
Comment (samypr100) [reply scenarios/excluded/dependency-excludes-non-contiguous-range-of-compatible-versions.toml:11]: The pre-existing `fork/*.toml` seem to also use this syntax, we can attempt to change it across though.

e.g. `fork/filter-sibling-dependencies.toml`
```
[packages.a.versions."4.3.0"]
[packages.a.versions."4.4.0"]
[packages.b.versions."1.0.0"]
requires = ["d==1.0.0"]
[packages.c.versions."1.0.0"]
requires = ["d==2.0.0"]
[packages.d.versions."1.0.0"]
[packages.d.versions."2.0.0"]
```

**Comment by konstin** [comment]
Separately from this PR, do we want to add prettier to this repo?

**Comment by zanieb** [comment]
I think @BurntSushi and @konstin will need to own deciding what's necessary to get this merged — I don't have strong feelings about the scenario format right now and they've been writing most of the recent scenarios.

**Comment by samypr100** [comment]
> I think @BurntSushi and @konstin will need to own deciding what's necessary to get this merged — I don't have strong feelings about the scenario format right now and they've been writing most of the recent scenarios.

👍 Should I rebase?

**Review by konstin — APPROVED** [review]
I like this and want to merge it

**Review by BurntSushi — APPROVED** [review]
Oh yes please! I found using TOML to write scenarios border-line pleasant.

**Comment by zanieb** [comment]
As discovered by @charliermarsh, it looks like this conversion dropped some information

e.g. https://github.com/astral-sh/packse/blob/2a670ba872d8ce6570765c3b89bc14788fdd982e/scenarios/requires-python.json#L231

is not present in the [toml version](https://github.com/astral-sh/packse/blob/main/scenarios/requires_python/incompatible-python-compatible-override.toml)

**Comment by notatallshaw** [comment]
This probably would have been a great time to do a round trip on the formats to proove no information loss, hindsight is 20-20 and all that.

**Comment by samypr100** [comment]
Apologies, not sure how I missed those. I opened https://github.com/astral-sh/packse/pull/226 to port those missing entries.

**Pull request merged**